### PR TITLE
feat: check node version in next config

### PR DESCRIPTION
**Pull request by c-ehrlich**
## ✅ Checklist

- [x] I have followed every step in the [contributing guide](https://github.com/t3-oss/create-t3-app/blob/main/CONTRIBUTING.md) (updated 2022-10-06).
- [x] The PR title follows the convention we established [conventional-commit](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] I performed a functional test on my final commit

---

would stop people from running into https://github.com/t3-oss/create-t3-app/issues/1600#issuecomment-1769189236

💯

### CODE REVIEW

**Comment by juliusmarminge** [comment]
I think 18.13 should be fine, if it's the same bug we had for uploadthing with the fetch api

File: cli/template/base/next.config.mjs [hunk 1]
```diff
@@ -20,3 +22,14 @@ const config = {
...
 
 export default config;
+
+function checkNodeVersion() {
+  const nodeVersion = process.versions.node;
+  const [major, minor] = nodeVersion.split(".").map((n) => parseInt(n));
+  if (!major || !minor) {
+    throw new Error(`Unable to parse Node version: ${nodeVersion}`);
+  }
+  if (major < 18 || (major === 18 && minor < 6) || (major === 19 && minor < 7)) {
```
Comment (sscotth) [inline cli/template/base/next.config.mjs:32]: should this be `minor < 16`?

Comment (juliusmarminge) [reply cli/template/base/next.config.mjs:32]: 16 is EOL

Comment (sscotth) [reply cli/template/base/next.config.mjs:32]: I think you misread `minor`, his check is "<18.6" not "<18.16"

Comment (c-ehrlich) [reply cli/template/base/next.config.mjs:32]: yes, thanks :)

**Comment by c-ehrlich** [comment]
> I think 18.13 should be fine, if it's the same bug we had for uploadthing with the fetch api
> 
> could we not just set the engines field in package json instead?

i dont think its possible to specify multiple versions? 20 is becoming LTR in a few days, so we could just require that, but would probably mean we also need to update docker docs etc

**Comment by juliusmarminge** [comment]
> > I think 18.13 should be fine, if it's the same bug we had for uploadthing with the fetch api
> > could we not just set the engines field in package json instead?
> 
> i dont think its possible to specify multiple versions? 20 is becoming LTR in a few days, so we could just require that, but would probably mean we also need to update docker docs etc

```json
"engines": {
  "node": "^18.16 || ^19.7 || >=20"
}
```

Doesnt' that work?

**Comment by sscotth** [comment]
Next just bumped their Node version to `>=18.17.0` in the v14.5.6 canary

https://github.com/vercel/next.js/pull/56896
https://github.com/vercel/next.js/pull/56943

**Pull request closed without merging**